elrond-wasm-debug/src/mandos_step/quick_call.rs

```diff
@@ -0,0 +1,72 @@
+use crate::{
+    num_bigint::BigUint,
+    tx_execution::{sc_call_with_async_and_callback, sc_query},
+    tx_mock::TxInput,
+    world_mock::BlockchainMock,
+    DebugApi,
+};
+use elrond_wasm::{
+    elrond_codec::{CodecFrom, PanicErrorHandler, TopEncodeMulti},
+    types::{Address, ContractCall, H256},
+};
+
+impl BlockchainMock {
+    pub fn quick_query<OriginalResult, RequestedResult>(
+        &mut self,
+        contract_call: ContractCall<DebugApi, OriginalResult>,
+    ) -> RequestedResult
+    where
+        OriginalResult: TopEncodeMulti,
+        RequestedResult: CodecFrom<OriginalResult>,
+    {
+        let tx_input = TxInput {
+            from: contract_call.to.to_address(),
+            to: contract_call.to.to_address(),
+            egld_value: BigUint::from(0u32),
+            esdt_values: Vec::new(),
+            func_name: contract_call.endpoint_name.to_boxed_bytes().into_vec(),
+            args: contract_call.arg_buffer.to_raw_args_vec(),
+            gas_limit: u64::MAX,
+            gas_price: 0u64,
+            tx_hash: H256::zero(),
+        };
+
+        let tx_result = self.with_borrowed_rc(|rc| sc_query(tx_input, rc.clone()));
+        assert!(tx_result.result_status == 0, "quick query failed"); // TODO: print more
+        assert!(
+            tx_result.result_status != 0 || tx_result.result_calls.is_empty(),
```

Comment:
This condition seems a bit weird. Shouldn't we only assert result calls? Since we should assert status == 0 for success.
```suggestion
            tx_result.result_calls.is_empty(),
```